main/app.c

```diff
@@ -2306,16 +2541,219 @@ int ast_app_group_update(struct ast_channel *old, struct ast_channel *new)
 	return 0;
 }
 
+/* Remove a channel from a group meta assignment */
+/* Right now this just removes all the group metadata for a group@category if this is the last channel in the group@category */
+/* Ideally we would have direct pointers from the channel group assignments into the metadata struct, so we don't have to search */
+int ast_app_group_remove_channel(struct ast_channel *chan, char *group, char *category)
```

Comment:
The name of this function is misleading since it doesn't actually remove the channel, it just does some extra logic tied to channel removal. I would either add the channel removal logic into this function, or rename it.

Reply:
Fixed